randovania/resolver/bootstrap.py

```diff
@@ -222,6 +227,53 @@ def assign_pool_results(self, rng: Random, patches: GamePatches, pool_results: P
             pool_results.starting
         )
 
+    def all_artifact_locations(
+        self,
+        game: GameDescription,
+        config: BaseConfiguration,
+        pre_placed_indices: list[PickupIndex],
+        game_specific_check: Callable[[PickupNode, BaseConfiguration], bool],
+    ) -> list[PickupNode]:
+        locations = []
+
+        for node in game.region_list.all_nodes:
+            if (
+                isinstance(node, PickupNode)
+                and node.pickup_index not in pre_placed_indices
+                and game_specific_check(node, config)
+            ):
+                locations.append(node)
+
+        return locations
+
+    def pre_place_artifacts(
+        self,
+        rng: Random,
+        patches: GamePatches,
+        pool_results: PoolResults,
+        game_specific_check: Callable[[PickupNode, BaseConfiguration], bool],
+        artifact_category: PickupCategory,
+    ) -> None:
+        pre_placed_indices = list(pool_results.assignment.keys())
+
+        locations = self.all_artifact_locations(
```

Comment:
I would prefer if we can pass an arbitrary list of locations here if desired

Reply:
Instead of passing the `game_specific_check` as argument?
To pass locations, ofc every game needs to create them.
Hmmm I don't want to copy paste the iteration over the nodes back to every game. 
I could copy paste a call to the common method `all_artifact_locations` to every game specific part and pass the result as argument but is there a reason for it?
I mean, it has to be a `PickupNode`, not an already assigned one by the preset and the rest logic comes from game specific argument. The only reason I see atm, that you maybe have a faster method to get the locations for one game instead of iterating over all nodes, which shouldn't be sooo important in the bootstrap part.

Reply:
the main use case I expect to be useful is when a game would distribute multiple such sets with virtually the same logic. consider e.g. a zelda game distributing dungeon map and compass within each dungeon

Reply:
The game specific part could just make multiple calls to `self.pre_place_artifacts` with a different check function. E.g. first call with indices of dungenon A and category of boss key dungeon A, second time with dungeon B etc.
Writing it down by myself, we may want to change it to specifiy the items instead of the item category in the future.

Anyways, I added a second commit to provide the locations instead of a check function but ofc it adds the boilerplate of calling `all_artifact_locations` to every game.